src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -21,8 +28,95 @@ public function getUrl(string $bucket, string $key): string
         return $this->getEndpoint($uri, [], null);
     }
 
-    /*
-     * TODO Add get() and put() which supports large files
-     * TODO add has()
+    public function download(string $bucket, string $key): ResultStream
+    {
+        return $this->getObject(['Bucket' => $bucket, 'Key' => $key])->getBody();
+    }
+
+    public function has(string $bucket, string $key): bool
+    {
+        return $this->objectExists(['Bucket' => $bucket, 'Key' => $key])->isSuccess();
+    }
+
+    /**
+     * @param string|resource|callable|iterable $object
+     * @param array{
+     *   ACL?: \AsyncAws\S3\Enum\ObjectCannedACL::*,
+     *   CacheControl?: string,
+     *   ContentType?: string,
+     *   Metadata?: array<string, string>,
+     * } $options
      */
+    public function upload(string $bucket, string $key, $object, array $options = []): void
+    {
+        // Send upload requests with size 64 MB
+        $chunkSize = 64 * 1024 * 1024;
```

Comment:
hmmm we are gonna put 64M of string in memory. isn't it too much ? 

should we create a temporary file, read few MB of chunks, and upload the file when reaching 64Mb? (or more)

Reply:
Im not sure if this is too much. The smaller this number is, the more requests are we doing. 

This is a "simple S3". If you want full control and optimize memory or requests, then you should use the `S3Client`. 

What is a good number for this?

Reply:
I use 32Mb.